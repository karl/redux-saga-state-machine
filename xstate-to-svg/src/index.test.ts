import * as fs from 'fs';
import * as mkdirp from 'mkdirp';
import { xstateToSvg } from './index';

const fixtures = [
  {
    name: 'simple',
    description: {
      key: 'light',
      initial: 'green',
      states: {
        green: {
          on: {
            TIMER: 'yellow',
            POWER_OUTAGE: 'red',
          },
        },
        yellow: {
          on: {
            TIMER: 'red',
            POWER_OUTAGE: 'red',
          },
        },
        red: {
          on: {
            TIMER: 'green',
            POWER_OUTAGE: 'red',
          },
        },
      },
    },
  },
  {
    name: 'nested',
    description: {
      key: 'light',
      initial: 'green',
      states: {
        green: {
          on: {
            TIMER: 'yellow',
          },
        },
        yellow: {
          on: {
            TIMER: 'red',
          },
        },
        red: {
          on: {
            TIMER: 'green',
          },
          initial: 'walk',
          states: {
            walk: {
              on: {
                PED_TIMER: 'wait',
              },
            },
            wait: {
              on: {
                PED_TIMER: 'stop',
              },
            },
            stop: {},
          },
        },
      },
    },
  },
];

describe('xstateToSvg', () => {
  fixtures.forEach((fixture) => {
    it(fixture.name, () => {
      const svg = xstateToSvg(fixture.description);

      mkdirp.sync(__dirname + '/images');
      fs.writeFileSync(__dirname + '/images/' + fixture.name + '.svg', svg);

      expect(svg).toMatchSnapshot();
    });
  });
});
