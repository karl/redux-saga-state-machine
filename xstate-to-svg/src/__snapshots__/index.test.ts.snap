// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xstateToSvg nested 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\"
 \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<!-- Generated by graphviz version 2.40.1 (20161225.0304)
 -->
<!-- Title: state transitions Pages: 1 -->
<svg width=\\"241pt\\" height=\\"396pt\\"
 viewBox=\\"0.00 0.00 241.00 396.00\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
<g id=\\"graph0\\" class=\\"graph\\" transform=\\"scale(1 1) rotate(0) translate(4 392)\\">
<title>state transitions</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" points=\\"-4,4 -4,-392 237,-392 237,4 -4,4\\"/>
<g id=\\"clust1\\" class=\\"cluster\\">
<title>cluster_red</title>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M20,-8C20,-8 92,-8 92,-8 98,-8 104,-14 104,-20 104,-20 104,-368 104,-368 104,-374 98,-380 92,-380 92,-380 20,-380 20,-380 14,-380 8,-374 8,-368 8,-368 8,-20 8,-20 8,-14 14,-8 20,-8\\"/>
<text text-anchor=\\"start\\" x=\\"47.8318\\" y=\\"-361.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">red</text>
</g>
<!-- initial -->
<g id=\\"node1\\" class=\\"node\\">
<title>initial</title>
<ellipse fill=\\"#000000\\" stroke=\\"#000000\\" stroke-width=\\"2\\" cx=\\"209\\" cy=\\"-324\\" rx=\\"5.5\\" ry=\\"5.5\\"/>
</g>
<!-- green -->
<g id=\\"node2\\" class=\\"node\\">
<title>green</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"233,-258 177,-258 177,-222 233,-222 233,-258\\"/>
<text text-anchor=\\"start\\" x=\\"189.661\\" y=\\"-236.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">green</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M189.3333,-223C189.3333,-223 220.6667,-223 220.6667,-223 226.3333,-223 232,-228.6667 232,-234.3333 232,-234.3333 232,-245.6667 232,-245.6667 232,-251.3333 226.3333,-257 220.6667,-257 220.6667,-257 189.3333,-257 189.3333,-257 183.6667,-257 178,-251.3333 178,-245.6667 178,-245.6667 178,-234.3333 178,-234.3333 178,-228.6667 183.6667,-223 189.3333,-223\\"/>
</g>
<!-- initial&#45;&gt;green -->
<g id=\\"edge1\\" class=\\"edge\\">
<title>initial&#45;&gt;green</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M208.7375,-318.4885C208.2581,-308.4198 207.2103,-286.4163 206.3458,-268.2615\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"209.8354,-267.9584 205.8636,-258.1362 202.8433,-268.2914 209.8354,-267.9584\\"/>
<text text-anchor=\\"middle\\" x=\\"208.3895\\" y=\\"-279\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\"> </text>
</g>
<!-- yellow -->
<g id=\\"node3\\" class=\\"node\\">
<title>yellow</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"187,-174 131,-174 131,-138 187,-138 187,-174\\"/>
<text text-anchor=\\"start\\" x=\\"142.335\\" y=\\"-152.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">yellow</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M143.3333,-139C143.3333,-139 174.6667,-139 174.6667,-139 180.3333,-139 186,-144.6667 186,-150.3333 186,-150.3333 186,-161.6667 186,-161.6667 186,-167.3333 180.3333,-173 174.6667,-173 174.6667,-173 143.3333,-173 143.3333,-173 137.6667,-173 132,-167.3333 132,-161.6667 132,-161.6667 132,-150.3333 132,-150.3333 132,-144.6667 137.6667,-139 143.3333,-139\\"/>
</g>
<!-- green&#45;&gt;yellow -->
<g id=\\"edge2\\" class=\\"edge\\">
<title>green&#45;&gt;yellow</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M195.0187,-221.7733C188.8099,-210.4355 180.7139,-195.6515 173.7705,-182.9723\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"176.8349,-181.281 168.9618,-174.1912 170.6952,-184.6433 176.8349,-181.281\\"/>
<text text-anchor=\\"start\\" x=\\"184\\" y=\\"-195\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">TIMER &#160;&#160;</text>
</g>
<!-- red -->
<!-- yellow&#45;&gt;red -->
<g id=\\"edge3\\" class=\\"edge\\">
<title>yellow&#45;&gt;red</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M150.7574,-174.0479C144.8543,-187.2237 136.8669,-205.5775 130.558,-222 122.0835,-244.0598 113.5533,-269.2001 107.088,-289.002\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"103.7502,-287.9489 103.9982,-298.5408 110.4095,-290.1061 103.7502,-287.9489\\"/>
<text text-anchor=\\"start\\" x=\\"130\\" y=\\"-237\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">TIMER &#160;&#160;</text>
</g>
<!-- red&#45;&gt;green -->
<g id=\\"edge4\\" class=\\"edge\\">
<title>red&#45;&gt;green</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M103.9953,-317.8385C118.4116,-306.7287 149.4832,-282.7836 173.3821,-264.3661\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"175.6817,-267.0127 181.4661,-258.1362 171.4087,-261.4681 175.6817,-267.0127\\"/>
<text text-anchor=\\"start\\" x=\\"157\\" y=\\"-279\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">TIMER &#160;&#160;</text>
</g>
<!-- walk -->
<g id=\\"node5\\" class=\\"node\\">
<title>walk</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"72,-342 16,-342 16,-306 72,-306 72,-342\\"/>
<text text-anchor=\\"start\\" x=\\"32.0018\\" y=\\"-320.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">walk</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M28.3333,-307C28.3333,-307 59.6667,-307 59.6667,-307 65.3333,-307 71,-312.6667 71,-318.3333 71,-318.3333 71,-329.6667 71,-329.6667 71,-335.3333 65.3333,-341 59.6667,-341 59.6667,-341 28.3333,-341 28.3333,-341 22.6667,-341 17,-335.3333 17,-329.6667 17,-329.6667 17,-318.3333 17,-318.3333 17,-312.6667 22.6667,-307 28.3333,-307\\"/>
</g>
<!-- wait -->
<g id=\\"node6\\" class=\\"node\\">
<title>wait</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"72,-174 16,-174 16,-138 72,-138 72,-174\\"/>
<text text-anchor=\\"start\\" x=\\"33.3344\\" y=\\"-152.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">wait</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M28.3333,-139C28.3333,-139 59.6667,-139 59.6667,-139 65.3333,-139 71,-144.6667 71,-150.3333 71,-150.3333 71,-161.6667 71,-161.6667 71,-167.3333 65.3333,-173 59.6667,-173 59.6667,-173 28.3333,-173 28.3333,-173 22.6667,-173 17,-167.3333 17,-161.6667 17,-161.6667 17,-150.3333 17,-150.3333 17,-144.6667 22.6667,-139 28.3333,-139\\"/>
</g>
<!-- walk&#45;&gt;wait -->
<g id=\\"edge5\\" class=\\"edge\\">
<title>walk&#45;&gt;wait</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M43.1048,-305.7964C42.4993,-292.5453 41.7551,-274.1653 41.442,-258 41.1321,-242.003 41.1321,-237.997 41.442,-222 41.6818,-209.6234 42.1742,-195.9486 42.6635,-184.2714\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"46.1635,-184.3472 43.1048,-174.2036 39.1702,-184.0407 46.1635,-184.3472\\"/>
<text text-anchor=\\"start\\" x=\\"42\\" y=\\"-237\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">PED_TIMER &#160;&#160;</text>
</g>
<!-- stop -->
<g id=\\"node7\\" class=\\"node\\">
<title>stop</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"72,-52 16,-52 16,-16 72,-16 72,-52\\"/>
<text text-anchor=\\"start\\" x=\\"32.6618\\" y=\\"-30.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">stop</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M28.3333,-17C28.3333,-17 59.6667,-17 59.6667,-17 65.3333,-17 71,-22.6667 71,-28.3333 71,-28.3333 71,-39.6667 71,-39.6667 71,-45.3333 65.3333,-51 59.6667,-51 59.6667,-51 28.3333,-51 28.3333,-51 22.6667,-51 17,-45.3333 17,-39.6667 17,-39.6667 17,-28.3333 17,-28.3333 17,-22.6667 22.6667,-17 28.3333,-17\\"/>
</g>
<!-- wait&#45;&gt;stop -->
<g id=\\"edge6\\" class=\\"edge\\">
<title>wait&#45;&gt;stop</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M44,-137.9985C44,-117.9958 44,-85.4427 44,-62.0831\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"47.5001,-62.0753 44,-52.0754 40.5001,-62.0754 47.5001,-62.0753\\"/>
<text text-anchor=\\"start\\" x=\\"44\\" y=\\"-92\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">PED_TIMER &#160;&#160;</text>
</g>
</g>
</svg>
"
`;

exports[`xstateToSvg simple 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\"
 \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<!-- Generated by graphviz version 2.40.1 (20161225.0304)
 -->
<!-- Title: state transitions Pages: 1 -->
<svg width=\\"321pt\\" height=\\"271pt\\"
 viewBox=\\"0.00 0.00 321.47 271.00\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
<g id=\\"graph0\\" class=\\"graph\\" transform=\\"scale(1 1) rotate(0) translate(4 267)\\">
<title>state transitions</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" points=\\"-4,4 -4,-267 317.4684,-267 317.4684,4 -4,4\\"/>
<!-- initial -->
<g id=\\"node1\\" class=\\"node\\">
<title>initial</title>
<ellipse fill=\\"#000000\\" stroke=\\"#000000\\" stroke-width=\\"2\\" cx=\\"167.0264\\" cy=\\"-257.5\\" rx=\\"5.5\\" ry=\\"5.5\\"/>
</g>
<!-- green -->
<g id=\\"node2\\" class=\\"node\\">
<title>green</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"195.0264,-204 139.0264,-204 139.0264,-168 195.0264,-168 195.0264,-204\\"/>
<text text-anchor=\\"start\\" x=\\"151.6874\\" y=\\"-182.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">green</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M151.3597,-169C151.3597,-169 182.6931,-169 182.6931,-169 188.3597,-169 194.0264,-174.6667 194.0264,-180.3333 194.0264,-180.3333 194.0264,-191.6667 194.0264,-191.6667 194.0264,-197.3333 188.3597,-203 182.6931,-203 182.6931,-203 151.3597,-203 151.3597,-203 145.6931,-203 140.0264,-197.3333 140.0264,-191.6667 140.0264,-191.6667 140.0264,-180.3333 140.0264,-180.3333 140.0264,-174.6667 145.6931,-169 151.3597,-169\\"/>
</g>
<!-- initial&#45;&gt;green -->
<g id=\\"edge1\\" class=\\"edge\\">
<title>initial&#45;&gt;green</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M167.0264,-251.8895C167.0264,-243.8095 167.0264,-228.335 167.0264,-214.4122\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"170.5265,-214.2159 167.0264,-204.2159 163.5265,-214.2159 170.5265,-214.2159\\"/>
<text text-anchor=\\"middle\\" x=\\"168.4159\\" y=\\"-225\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\"> </text>
</g>
<!-- yellow -->
<g id=\\"node3\\" class=\\"node\\">
<title>yellow</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"88.0264,-120 32.0264,-120 32.0264,-84 88.0264,-84 88.0264,-120\\"/>
<text text-anchor=\\"start\\" x=\\"43.3614\\" y=\\"-98.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">yellow</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M44.3597,-85C44.3597,-85 75.6931,-85 75.6931,-85 81.3597,-85 87.0264,-90.6667 87.0264,-96.3333 87.0264,-96.3333 87.0264,-107.6667 87.0264,-107.6667 87.0264,-113.3333 81.3597,-119 75.6931,-119 75.6931,-119 44.3597,-119 44.3597,-119 38.6931,-119 33.0264,-113.3333 33.0264,-107.6667 33.0264,-107.6667 33.0264,-96.3333 33.0264,-96.3333 33.0264,-90.6667 38.6931,-85 44.3597,-85\\"/>
</g>
<!-- green&#45;&gt;yellow -->
<g id=\\"edge2\\" class=\\"edge\\">
<title>green&#45;&gt;yellow</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M140.8,-167.911C132.9145,-162.3047 124.3081,-156.0166 116.5844,-150 107.25,-142.7287 97.331,-134.4942 88.4492,-126.9243\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"90.4962,-124.0686 80.6305,-120.206 85.9341,-129.3778 90.4962,-124.0686\\"/>
<text text-anchor=\\"start\\" x=\\"117.0264\\" y=\\"-141\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">TIMER &#160;&#160;</text>
</g>
<!-- red -->
<g id=\\"node4\\" class=\\"node\\">
<title>red</title>
<polygon fill=\\"#ffffff\\" stroke=\\"transparent\\" stroke-width=\\"2\\" points=\\"141.0264,-36 85.0264,-36 85.0264,0 141.0264,0 141.0264,-36\\"/>
<text text-anchor=\\"start\\" x=\\"104.3582\\" y=\\"-14.2\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"12.00\\" fill=\\"#000000\\">red</text>
<path fill=\\"none\\" stroke=\\"#000000\\" stroke-width=\\"2\\" d=\\"M97.3597,-1C97.3597,-1 128.6931,-1 128.6931,-1 134.3597,-1 140.0264,-6.6667 140.0264,-12.3333 140.0264,-12.3333 140.0264,-23.6667 140.0264,-23.6667 140.0264,-29.3333 134.3597,-35 128.6931,-35 128.6931,-35 97.3597,-35 97.3597,-35 91.6931,-35 86.0264,-29.3333 86.0264,-23.6667 86.0264,-23.6667 86.0264,-12.3333 86.0264,-12.3333 86.0264,-6.6667 91.6931,-1 97.3597,-1\\"/>
</g>
<!-- green&#45;&gt;red -->
<g id=\\"edge3\\" class=\\"edge\\">
<title>green&#45;&gt;red</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M169.5039,-167.7383C172.2374,-141.1134 174.0215,-90.8078 155.0264,-54 153.094,-50.2556 150.5904,-46.7372 147.768,-43.4774\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"150.2438,-41.0034 140.7228,-36.3557 145.2673,-45.9263 150.2438,-41.0034\\"/>
<text text-anchor=\\"start\\" x=\\"171.0264\\" y=\\"-99\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">POWER_OUTAGE &#160;&#160;</text>
</g>
<!-- yellow&#45;&gt;red -->
<g id=\\"edge4\\" class=\\"edge\\">
<title>yellow&#45;&gt;red</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M32.0131,-91.6354C12.7539,-82.871 -7.5022,-69.3179 3.5844,-54 12.1752,-42.1305 46.7778,-32.0559 74.8056,-25.6093\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"75.7671,-28.9813 84.7738,-23.4018 74.2536,-22.1469 75.7671,-28.9813\\"/>
<text text-anchor=\\"start\\" x=\\"4.0264\\" y=\\"-57\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">TIMER &#160;&#160;</text>
</g>
<!-- yellow&#45;&gt;red -->
<g id=\\"edge5\\" class=\\"edge\\">
<title>yellow&#45;&gt;red</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M57.1463,-83.6564C56.5505,-74.2532 57.2571,-62.9331 62.1314,-54 65.6598,-47.5336 70.9095,-41.9609 76.716,-37.2639\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"78.7947,-40.08 84.8695,-31.3997 74.7075,-34.3972 78.7947,-40.08\\"/>
<text text-anchor=\\"start\\" x=\\"63.0264\\" y=\\"-57\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">POWER_OUTAGE &#160;&#160;</text>
</g>
<!-- red&#45;&gt;green -->
<g id=\\"edge6\\" class=\\"edge\\">
<title>red&#45;&gt;green</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M141.2819,-20.8175C177.7331,-26.1389 239.8004,-41.3029 268.0264,-84 290.7679,-118.4008 241.9093,-150.9468 204.5123,-169.6623\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"202.7816,-166.6114 195.3021,-174.1152 205.8285,-172.9135 202.7816,-166.6114\\"/>
<text text-anchor=\\"start\\" x=\\"274.0264\\" y=\\"-99\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">TIMER &#160;&#160;</text>
</g>
<!-- red&#45;&gt;red -->
<g id=\\"edge7\\" class=\\"edge\\">
<title>red&#45;&gt;red</title>
<path fill=\\"none\\" stroke=\\"#000000\\" d=\\"M141.2926,-25.8269C149.0774,-25.4488 155.0264,-22.8398 155.0264,-18 155.0264,-15.5801 153.5392,-13.7179 151.0788,-12.4134\\"/>
<polygon fill=\\"#000000\\" stroke=\\"#000000\\" points=\\"151.8215,-8.9929 141.2926,-10.1731 150.2594,-15.8164 151.8215,-8.9929\\"/>
<text text-anchor=\\"start\\" x=\\"155.0264\\" y=\\"-15\\" font-family=\\"Helvetica,sans-Serif\\" font-size=\\"10.00\\" fill=\\"#000000\\">POWER_OUTAGE &#160;&#160;</text>
</g>
</g>
</svg>
"
`;
